File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.java
class Solution {
    public String gcdOfStrings(String str1, String str2) {
        if(!(str1 + str2).equals(str2 + str1)){
            return "";
        }
        int x = gcd(str1.length(), str2.length());
        return str1.substring(0,x);
    }
    public static int gcd(int a, int b){
        if(b==0){
            return a;
        }
        else{
           return gcd(b,a%b);
        }
    }
}

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if(str1+str2 != str2+str1):
            return ""
        x = gcd(len(str1),len(str2))
        return str1[:x]
    
    def gcd(self, a:int, b:int) -> int:
        return a if b==0 else gcd(b, a%b)